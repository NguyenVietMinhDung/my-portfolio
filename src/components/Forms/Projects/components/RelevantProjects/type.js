import type { Props as Item } from '../RelevantProjectItem/type';

export type Props = {
  items: Array<Item>,
};
