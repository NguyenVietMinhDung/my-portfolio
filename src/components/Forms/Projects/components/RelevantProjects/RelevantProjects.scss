.relevant-projects {
	display: flex;
	flex-flow: column wrap;
	justify-content: center;
}
