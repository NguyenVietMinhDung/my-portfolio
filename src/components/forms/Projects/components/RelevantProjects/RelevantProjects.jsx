// @flow
import React, { memo } from 'react';
import CSSModules from 'react-css-modules';
import styles from './RelevantProjects.scss';
import type { Props } from './type';
import RelevantProjectItem from '../RelevantProjectItem';

const RelevantProjects = (props: Props) => {
  const { items } = props;
  return (
    <ul styleName="relevant-projects">
      {items.map(item => (
        <RelevantProjectItem
          key={item.name}
          {...item}
        />
      ))}
    </ul>
  );
};

export default memo<Props>(CSSModules(RelevantProjects, styles));
