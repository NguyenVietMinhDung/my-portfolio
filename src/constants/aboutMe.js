export const TITLE = 'About Me';
export const GREETING = 'Hi There! I\'m Dung Nguyen';
export const SUMMARY = 'I am an ambitious Web\n'
  + 'Application Developer who\n'
  + 'always tries to improve himself\n'
  + 'everyday, a quick learner who\n'
  + 'can absorb new ideas and can\n'
  + 'communicate clearly and\n'
  + 'effectively. Looking for an\n'
  + 'ambitious company which will\n'
  + 'challenge my developer and\n'
  + 'allows me to continue\n'
  + 'to develop my knowledge and\n'
  + 'potential.';
export const basicInfo = [
  {
    iconClassName: 'lni-calendar',
    name: 'Date of birth',
    value: '16 January 1995',
  },
  {
    iconClassName: 'lni-envelope',
    name: 'Email',
    value: 'ngvietminhdung@gmail.com',
  },
  {
    iconClassName: 'lni-phone-handset',
    name: 'Phone',
    value: '(076) 401-2912',
  },
  {
    iconClassName: 'lni-map-marker',
    name: 'Location',
    value: 'Tan Binh District, HCM City',
  },
];
export const skills = [
  {
    itemId: 'skill-1',
    name: 'JavaScript, HTML, CSS',
    value: 80,
  },
  {
    itemId: 'skill-2',
    name: 'ReactJS',
    value: 60,
  },
  {
    itemId: 'skill-3',
    name: 'SCSS',
    value: 60,
  },
  {
    itemId: 'skill-5',
    name: 'Webpack',
    value: 50,
  },
  {
    itemId: 'skill-6',
    name: 'Node.js',
    value: 60,
  },
  {
    itemId: 'skill-7',
    name: 'Git',
    value: 60,
  },
];
