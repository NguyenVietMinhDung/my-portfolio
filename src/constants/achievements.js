export const TITLE = 'Achievements';
export const timeline = {
  iconClassName: 'lni-certificate',
  items: [
    {
      key: 1,
      organization: 'Amazon Web Services (AWS)',
      startDate: 'August 2020',
      endDate: 'August 2023',
      description: 'AWS Certified Cloud Practitioner',
    },
  ],
};
