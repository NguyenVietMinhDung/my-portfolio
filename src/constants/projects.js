export const TITLE = 'Relevant Projects';
export const items = [
  {
    isCurrentProject: false,
    name: 'Textme',
    role: 'Developer',
    startDate: 'September 2019',
    endDate: 'May 2020',
    description: 'A chat tool that helps customers can contact with bankers and versa vice',
  },
  {
    isCurrentProject: false,
    name: 'Huddle',
    role: 'Developer',
    startDate: 'January 2019',
    endDate: 'June 2019',
    description: 'A ticket-selling website for US high school events',
  },
  {
    isCurrentProject: false,
    name: 'ERTPMTool',
    role: 'Developer',
    startDate: 'July 2018',
    endDate: 'January 2019',
    description: 'A micro-services project management application\n'
    + 'which helps managers to keep track of resource allocation',
  },
  {
    isCurrentProject: false,
    name: 'PDV',
    role: 'Developer',
    startDate: 'February 2018',
    endDate: 'June 2018',
    description: 'Provider Search is a user interface that allows\n'
    + 'users to search for practitioners in the MasterFile. Built on the Single',
  },
  {
    isCurrentProject: false,
    name: 'DMTool',
    role: 'Developer',
    startDate: 'September 2017',
    endDate: 'February 2018',
    description: 'An internal tool which supports managers to keep track of resource allocation',
  },
];
