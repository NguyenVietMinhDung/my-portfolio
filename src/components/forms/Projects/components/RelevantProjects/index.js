import RelevantProjects from './RelevantProjects';

export default RelevantProjects;
