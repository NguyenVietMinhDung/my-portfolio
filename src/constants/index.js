import * as actions from './actions';
import * as app from './app';
import * as navbar from './navbar';
import * as header from './header';
import * as footer from './footer';
import * as aboutMe from './aboutMe';
import * as experiences from './experiences';
import * as educations from './educations';
import * as projects from './projects';
import * as achievements from './achievements';

export default {
  actions,
  app,
  navbar,
  header,
  footer,
  aboutMe,
  experiences,
  educations,
  projects,
  achievements,
};
