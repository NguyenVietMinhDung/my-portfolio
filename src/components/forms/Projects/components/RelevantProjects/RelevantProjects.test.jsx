import React from 'react';
import { shallow } from 'enzyme';
import { items } from '../../../../../constants/projects';
import RelevantProjects from './RelevantProjects';

const defaultProps = {
  items,
};

describe('<RelevantProjects />', () => {
  test('renders correctly', () => {
    const component = shallow(<RelevantProjects {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
