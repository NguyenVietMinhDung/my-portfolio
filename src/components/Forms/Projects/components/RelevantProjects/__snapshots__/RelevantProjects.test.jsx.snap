// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RelevantProjects /> renders correctly 1`] = `
<ul
  className="relevant-projects"
>
  <Memo(RelevantProjectItem)
    description="A chat tool that helps customers can contact with bankers and versa vice"
    endDate="May 2020"
    isCurrentProject={false}
    key="Textme"
    name="Textme"
    role="Developer"
    startDate="September 2019"
  />
  <Memo(RelevantProjectItem)
    description="A ticket-selling website for US high school events"
    endDate="June 2019"
    isCurrentProject={false}
    key="Huddle"
    name="Huddle"
    role="Developer"
    startDate="January 2019"
  />
  <Memo(RelevantProjectItem)
    description="A micro-services project management application
which helps managers to keep track of resource allocation"
    endDate="January 2019"
    isCurrentProject={false}
    key="ERTPMTool"
    name="ERTPMTool"
    role="Developer"
    startDate="July 2018"
  />
  <Memo(RelevantProjectItem)
    description="Provider Search is a user interface that allows
users to search for practitioners in the MasterFile. Built on the Single"
    endDate="June 2018"
    isCurrentProject={false}
    key="PDV"
    name="PDV"
    role="Developer"
    startDate="February 2018"
  />
  <Memo(RelevantProjectItem)
    description="An internal tool which supports managers to keep track of resource allocation"
    endDate="February 2018"
    isCurrentProject={false}
    key="DMTool"
    name="DMTool"
    role="Developer"
    startDate="September 2017"
  />
</ul>
`;
